components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/DefaultAttributeFinder.java

```diff
@@ -60,7 +71,46 @@ public String getModuleName() {
         return "Default Attribute Finder";
     }
 
-    /*
+    /**
+     * This method is introduced in order to check whether the user is a local or federated and if it is a
+     * federated user, prevent from obtaining user attributes from userstore.
+     * @param attributeType
+     * @param attributeId The unique id of the required attribute.
+     * @param category  The category of the required attribute.
+     * @param issuer The attribute issuer.
+     * @param evaluationCtx The evaluation context object.
+     * @return return the set of values for the required attribute.
+     * @throws Exception throws if fails.
+     */
+    @Override
+    public Set<String> getAttributeValues(URI attributeType, URI attributeId, URI category,
+                                          String issuer, EvaluationCtx evaluationCtx) throws Exception {
+
+        Set<String> values = null;
+        EvaluationResult userType = evaluationCtx.getAttribute(new URI(StringAttribute.identifier), new URI(
+                PDPConstants.USER_TYPE_ID), issuer, new URI(PDPConstants.USER_CATEGORY));
+        String userTypeId = null;
+        if (userType != null && userType.getAttributeValue() != null && userType.getAttributeValue().isBag()) {
+            BagAttribute bagAttribute = (BagAttribute) userType.getAttributeValue();
+            if (bagAttribute.size() > 0) {
+                userTypeId = ((AttributeValue) bagAttribute.iterator().next()).encode();
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("The user is a %1$s user", userTypeId));
```

Comment:
Is it possible to include the username in the log ?

Reply:
It is better to have following to avoid confusion in the log
String.format("The user type of the user is %s", userTypeId)
otherwise the log is printed like
```
The user is a locals user
```